esp/esp/users/forms/user_profile.py

```diff
@@ -339,14 +344,22 @@ def clean(self):
         super(TeacherInfoForm, self).clean()
         cleaned_data = self.cleaned_data
 
-        # If teacher is not from MIT, make sure they've filled in the next box
-        from_here = cleaned_data.get('from_here')
-        school = cleaned_data.get('school')
-
-        if from_here == "False" and school == "":
-            msg = u'Please enter your affiliation if you are not from %s.' % settings.INSTITUTION_NAME
-            self.add_error('school', msg)
-
+        affiliation_field = self.fields['affiliation']
+        affiliation_with_data = affiliation_field.widget.decompress(cleaned_data.get('affiliation'))
```

Comment:
if both of the elements are guaranteed to exist, even if empty string, you can do `affiliation, school = ...` which might be cleaner.